src/main/java/org/apache/sling/discovery/impl/DiscoveryServiceConfig.java

```diff
@@ -0,0 +1,138 @@
+    int connectionTimeout() default DEFAULT_CONNECTION_TIMEOUT;
+
+    /** Configure the socket read timeout (SO_TIMEOUT) for topology connectors. */
+    int DEFAULT_SO_TIMEOUT = 10;
+    @AttributeDefinition(name = "%soTimeout.name", description = "%soTimeout.description")
+    int soTimeout() default DEFAULT_SO_TIMEOUT;
+
+    /** Name of the repository descriptor to be taken into account for leader election:
+     those instances have preference to become leader which have the corresponding descriptor value of 'false' */
+    @AttributeDefinition(name = "%leaderElectionRepositoryDescriptor.name", description = "%leaderElectionRepositoryDescriptor.description")
+    String leaderElectionRepositoryDescriptor();
+
+    /**
+     * Whether or not (default false) the leaderElectionRepositoryDescriptor should be inverted (if that one
+     * is configured at all).
+     */
+    @AttributeDefinition(name = "%invertRepositoryDescriptor.name", description = "%invertRepositoryDescriptor.description")
+    boolean invertRepositoryDescriptor() default false;
+
+    /** URLs where to join a topology, eg http://localhost:4502/libs/sling/topology/connector */
+    @AttributeDefinition(name = "%topologyConnectorUrls.name", description = "%topologyConnectorUrls.description", cardinality=1024)
+    String[] topologyConnectorUrls() default {};
+
+    /** list of ips and/or hostnames which are allowed to connect to /libs/sling/topology/connector */
+    @AttributeDefinition(name = "%topologyConnectorWhitelist.name", description = "%topologyConnectorWhitelist.description")
+    String[] topologyConnectorWhitelist() default {"localhost", "127.0.0.1"};
+    String[] DEFAULT_TOPOLOGY_CONNECTOR_WHITELIST = {"localhost", "127.0.0.1"};
+
+    /** Path of resource where to keep discovery information, e.g /var/discovery/impl/ */
+    @AttributeDefinition(name = "%discoveryResourcePath.name", description = "%discoveryResourcePath.description")
+    String discoveryResourcePath() default DEFAULT_DISCOVERY_RESOURCE_PATH;
+    String DEFAULT_DISCOVERY_RESOURCE_PATH = "/var/discovery/impl/";
+
+    /**
+     * If set to true, local-loops of topology connectors are automatically stopped when detected so.
+     */
+    @AttributeDefinition(name = "%autoStopLocalLoopEnabled.name", description = "%autoStopLocalLoopEnabled.description")
+    boolean autoStopLocalLoopEnabled() default false;
+
+    /**
+     * If set to true, request body will be gzipped - only works if counter-part accepts gzip-requests!
+     */
+    @AttributeDefinition(name = "%gzipConnectorRequestsEnabled.name", description = "%gzipConnectorRequestsEnabled.description")
+    boolean gzipConnectorRequestsEnabled() default false;
+
+    /**
+     * If set to true, hmac is enabled and the white list is disabled.
+     */
+    @AttributeDefinition(name = "%hmacEnabled.name", description = "%hmacEnabled.description")
+    boolean hmacEnabled() default true;
```

Comment:
I set the default to true, because earlier the default already seemed to be true:
https://github.com/apache/sling-org-apache-sling-discovery-impl/blob/de269f64678b9a41693f83ddbd63bc98bad9d966/src/main/java/org/apache/sling/discovery/impl/Config.java#L332